src/test/app/Batch_test.cpp

```diff
@@ -0,0 +1,3047 @@
+                    0,
+                    aliceSeq + 1),  // Uses Ticket (Fails with terPRE_TICKET)
+                batch::inner(pay(alice, bob, XRP(2)), aliceSeq + 2));
+            auto const noopTxn = env.jt(noop(alice), seq(aliceSeq + 1));
+            env(noopTxn, ter(terPRE_SEQ));
+            env(batchTxn);
+            env.close();
+
+            auto const txIDs = batchTxn.stx->getBatchTransactionIDs();
+            TxID const parentBatchId = batchTxn.stx->getTransactionID();
+            Json::Value const jrr = getLastLedger(env);
+            std::vector<TestLedgerData> testCases = {
+                {0, "Batch", "tesSUCCESS", to_string(parentBatchId), false},
+                {1,
+                 "AccountSet",
+                 "tesSUCCESS",
+                 to_string(noopTxn.stx->getTransactionID()),
+                 false},
+            };
+            validateLedgerTxns(env, jrr, testCases, parentBatchId);
+        }
+
+        // Before Batch Txn w/ same sequence
+        {
+            test::jtx::Env env{*this, envconfig()};
+            env.fund(XRP(1000), alice, bob);
+            env.close();
+
+            auto const aliceSeq = env.seq(alice);
+            auto const batchFee = batch::calcBatchFee(env, 0, 2);
+            auto const batchTxn = env.jt(
+                batch::outer(alice, aliceSeq, batchFee, tfAllOrNothing),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq + 1),
+                batch::inner(pay(alice, bob, XRP(2)), aliceSeq + 2));
+            auto const noopTxn = env.jt(noop(alice), seq(aliceSeq + 1));
+            env(noopTxn, ter(terPRE_SEQ));
+            env(batchTxn);
+            env.close();
+
+            auto const txIDs = batchTxn.stx->getBatchTransactionIDs();
+            TxID const parentBatchId = batchTxn.stx->getTransactionID();
+            Json::Value const jrr = getLastLedger(env);
+            std::vector<TestLedgerData> testCases = {
+                {0, "Batch", "tesSUCCESS", to_string(parentBatchId), false},
+                {1, "Payment", "tesSUCCESS", to_string(txIDs[0]), true},
+                {2, "Payment", "tesSUCCESS", to_string(txIDs[1]), true},
+            };
+            validateLedgerTxns(env, jrr, testCases, parentBatchId);
+        }
+
+        // After Batch Txn w/ same sequence
+        {
+            test::jtx::Env env{*this, envconfig()};
+            env.fund(XRP(1000), alice, bob);
+            env.close();
+
+            auto const aliceSeq = env.seq(alice);
+            auto const batchFee = batch::calcBatchFee(env, 0, 2);
+            auto const batchTxn = env.jt(
+                batch::outer(alice, aliceSeq, batchFee, tfAllOrNothing),
+                batch::inner(pay(alice, bob, XRP(1)), aliceSeq + 1),
+                batch::inner(pay(alice, bob, XRP(2)), aliceSeq + 2));
+            auto const noopTxn = env.jt(noop(alice), seq(aliceSeq + 1));
+            env(batchTxn);
+            env(noopTxn);
```

Comment:
I'm confused - why does this transaction succeed?

Reply:
From my understanding, this is on the open ledger, since inner transactions aren't processed, AccountSet is successful despite using bad sequence. But during validation, AccountSet fails due to it's sequence conflicting with the inner txns, which is why testCases doesn't include the AccountSet tx